### PR TITLE
refactor(core): Standardize db table names 

### PR DESCRIPTION
**Pull request by RicardoE105**
## Summary

Make all table names singular.

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/cli/src/databases/entities/event-destinations.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { MessageEventBusDestinationOptions } from 'n8n-workflow';
 
 import { WithTimestamps, jsonColumnType } from './abstract-entity';
 
-@Entity({ name: 'event_destinations' })
+@Entity({ name: 'event_destination' })
 export class EventDestinations extends WithTimestamps {
```
Comment (RicardoE105) [inline packages/cli/src/databases/entities/event-destinations.ts:7]: in another PR will make the class names singular!

File: packages/cli/src/controllers/e2e.controller.ts [hunk 2]
```diff
@@ -29,23 +29,23 @@ if (!inE2ETests) {
 const tablesToTruncate = [
 	'auth_identity',
 	'auth_provider_sync_history',
-	'credentials_entity',
+	'credentials',
```
Comment (tomi) [inline packages/cli/src/controllers/e2e.controller.ts:32]:
```suggestion
	'credential',
```

File: packages/cli/src/controllers/e2e.controller.ts [hunk 3]
```diff
@@ -29,23 +29,23 @@ if (!inE2ETests) {
...
 	'event_destinations',
```
Comment (tomi) [inline packages/cli/src/controllers/e2e.controller.ts:33]:
```suggestion
	'event_destination',
```

File: packages/cli/src/controllers/e2e.controller.ts [hunk 4]
```diff
@@ -29,23 +29,23 @@ if (!inE2ETests) {
...
+	'tag',
 	'user',
 	'variables',
-	'webhook_entity',
-	'workflow_entity',
+	'webhook',
+	'workflow',
 	'workflow_statistics',
-	'workflows_tags',
+	'workflows_tag',
```
Comment (tomi) [inline packages/cli/src/controllers/e2e.controller.ts:48]:
```suggestion
	'workflow_tag',
```

File: packages/cli/src/controllers/e2e.controller.ts [hunk 5]
```diff
@@ -29,23 +29,23 @@ if (!inE2ETests) {
...
 	'project',
 	'project_relation',
 	'settings',
-	'shared_credentials',
+	'shared_credential',
 	'shared_workflow',
-	'tag_entity',
+	'tag',
 	'user',
 	'variables',
```
Comment (tomi) [inline packages/cli/src/controllers/e2e.controller.ts:44]:
```suggestion
	'variable',
```

File: packages/cli/src/databases/migrations/common/1727369628621-StandardizeTableNames.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+			tag_entity: 'tag',
+			user_api_keys: 'user_api_key',
+			variables: 'variable',
+			webhook_entity: 'webhook',
+			workflow_entity: 'workflow',
+			workflows_tags: 'workflow_tag',
+		} as const;
+
+		await Promise.all(
+			// eslint-disable-next-line @typescript-eslint/promise-function-async
```
Comment (tomi) [inline packages/cli/src/databases/migrations/common/1727369628621-StandardizeTableNames.ts:25]: You can remove this rule disable if you mark the function inside map as async

File: packages/cli/src/controllers/e2e.controller.ts (hunk 2 above)
Comment (RicardoE105) [reply packages/cli/src/controllers/e2e.controller.ts:32]: will update them, but considering that the tests did not fail makes me think truncating this table it's not neccesary

File: packages/cli/src/databases/migrations/common/1727369628621-StandardizeTableNames.ts (hunk 6 above)
Comment (RicardoE105) [reply packages/cli/src/databases/migrations/common/1727369628621-StandardizeTableNames.ts]: Did that but then got a different error, which is an unsafe return type. Switching to runQuery seems to fix the issue,

CleanShot 2024-10-01 at 16 38 24@2x

CleanShot 2024-10-01 at 16 39 29@2x

**Pull request closed without merging**